Nodetools docs improvements 1/N

denesb opened the pull request:
While working on https://github.com/scylladb/scylladb/issues/15588, I noticed problems with the existing documentation, when comparing it with the actual code.
This PR contains fixes for nodetool compact, stop and scrub.

denesb commented on docs/operating-scylla/nodetool-commands/compact.rst:
```diff
@@ -15,28 +15,12 @@ Syntax
-* ``--`` - Separates command-line options from the list of argument(useful when an argument might be mistaken for a command-line option).
-
-The following options are NOT supported:
+The following options are accepted, but not implemented:
```
According to @nyh just saying "not supported" is ambiguous, because it can also simply mean options which are not recognized by the tool, like `--dog` or `--cat`.

These options are present in C* nodetool implementation, but the ScyllaDB backend doesn't implement the functionality attached to them, so they are no-op, they are silenty accepted and ignored.

denesb replied:
> By the way, I think we shouldn't silently ignore any option... Even if it was supported in the past by Cassandra... Another question is whether these options are **really** silently ignored, or something else happens. I remember @denesb saying that they aren't ignored, but rather used as a keyspace name (!).

This was the case for an option which was not supported by our Java nodetool either. I guess the parameter made it into our documentation because of copy-pasta from a more recent version of C* nodetool's doc, which does support this parameter.

nyh replied:
> > After all, the "--dog" option also ins't supported - why didn't we list it as well? ;-)
> 
> This is a very good question. I've opposed more than once against listing what is NOT supported, as, in general, it's a bad documentation practice. However, I was told that some information might be important for people familiar with Cassandra, and we need to point out the differences. Since we don't know for sure if the options are silently ignored

We need to know the answer before writing the documentation :-) I'm sure @denesb knows, or can check:

> how about saying:
> 
> _The following options available in Cassandra are NOT implemented in ScyllaDB:_
> 
> @nyh What do you think?

I think we should mention something like "Cassandra's Nodetool", not "Cassandra". What we say about "Cassandra" actually applies also to Scylla, if Cassandra's Java version of nodetool is used for it. I don't know if we have better names than "Cassandra's nodetool" vs "ScyllaDB's nodetool".

denesb replied:
> We need to know the answer before writing the documentation :-) I'm sure @denesb knows, or can check

It varies case-by-case. Most of the ones I checked so far are silently ignore, but some cause failures. In all the cases I try to write tests that document and enshrine the current behaviour.

denesb commented:
Update:
* Rebased
* Dropped commit about scrub procedure, I will instead add a separate page dedicated to scrub and will add a link to it on nodetool scrub
* Changed phrase about unsupported options

denesb commented:
@scylladb/scylla-maitn ping.

denesb commented:
@scylladb/scylla-maint ping^2.

avikivity commented:
dequeued due to terrible state of next

avikivity commented:
requeued
